handler/interceptors.go

```diff
@@ -375,6 +377,43 @@ func NoOpInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamSer
 	return handler(srv, ss)
 }
 
+// NoOpInterceptor is a gRPC interceptor which doesn't do payment checking.
+func AllowedUserValidationInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo,
+	handler grpc.StreamHandler) error {
+	context, err := getGrpcContext(ss, info)
+	if err != nil {
+		return err.Err()
+	}
+	if err := checkIfSignerIsAnAllowedUser(context); err != nil {
+		return err
+	}
+	return handler(srv, ss)
+}
+
+func checkIfSignerIsAnAllowedUser(context *GrpcStreamContext) (err error) {
+	signature, grpcErr := GetBytes(context.MD, AllowedUserSignatureHeader)
```

Comment:
I would also add separate `snet-payment-type` value for such type of authorization in order to have consistent metadata format for all cases. Thus `snet-payment-type` can always be used to understand which authorization procedure is applicable to the header and which fields are available.